### PR TITLE
Update Display Names - Pokémon Forms (lowercase) -> (Uppercase)

### PR DESCRIPTION
**Pull request by CorgiOnNeptune**
Part 3/3 of "fixing" display names to what is "proper" canon capitalization and looks cleaner imo.

This one is the biggest with updating all the Pokémon form names to use "proper" capitalization in their brackets. Was pretty thorough, so I'm confident I didn't miss anything important. But I guess it's always possible.. At least there are no errors in the game opening and playing with it a little bit. 

E.g. `Bulbasaur (clone)` -> `Bulbasaur (Clone)`; `Pikachu (Hoenn cap)` -> `Pikachu (Hoenn Cap)`; etc.

Also updated the comment on the `PokemonNameType.ts` file.

### CODE REVIEW

**Comment by Jaaslet** [comment]
> Isn't this just changing a few strings?
> 
> I don't think it'll go wrong.

What's your point?

**Comment by Bassem-Const** [comment]
> > Isn't this just changing a few strings?
> > I don't think it'll go wrong.
> 
> What's your point?

You aren't adding any new code, no new variables.

**Comment by Bassem-Const** [comment]
@CorgiOnNeptune
UHOH, THIS BRANCH IS OUT OF DATE

**Comment by Bassem-Const** [comment]
> > You aren't adding any new code, no new variables.
> 
> Maybe? I have not looked at it yet?

I don't think you need to change any values. @Jaaslet

**Comment by Qwertypop04** [comment]
> @CorgiOnNeptune
> UHOH, THIS BRANCH IS OUT OF DATE

Okay? That happens when other pull requests are merged, it's not a big deal at all.

**Comment by CorgiOnNeptune** [comment]
> Isn't this just changing a few strings?
> 
> I don't think it'll go wrong.

I just found something I missed that would've broken a couple shops and a town. Hence the little bit of concern in my OP. But match case search function is good, so now there's nothing I missed for sure.

File: src/scripts/Update.ts [hunk 1]
```diff
@@ -916,6 +916,98 @@ class Update implements Saveable {
...
+                [35, '100 x X Click'],
+                [40, '100 x X Attack'],
+                [50, '100 x Small Restore'],
+                [100, 'Deoxys'],
+                [110, '10 x Water Stone'],
+                [120, '10 x Leaf Stone'],
+                [130, '10 x Thunder Stone'],
+                [140, '10 x Fire Stone'],
+                [150, '200 x Medium Restore'],
+                [151, 'Deoxys (Attack)'],
```
[the remark this answers is not in the document]
Comment (Jaaslet) [reply src/scripts/Update.ts:935]: And for some reason, the update function does not remove 'Deoxys (Attack)', but it works for the rest

**Review by Jaaslet — CHANGES_REQUESTED** [review]
Looks good. A couple of them looked weird, but i looked them up, and they are all correct.
Let's try to merge this quickly, to prevent too many merge conflicts. I just have some changes to the update function:

File: src/scripts/Update.ts [hunk 2]
```diff
@@ -916,6 +916,98 @@ class Update implements Saveable {
...
+            ];
+            const highestStageCompleted = saveData.statistics?.battleFrontierHighestStageCompleted || 0;
+            saveData.battleFrontier = {
+                milestones: milestones.filter(([stage]) => stage <= highestStageCompleted),
+                checkpoint: saveData.battleFrontier.checkpoint,
+            };
+
+            // Update Pokemon name changes
+            const renamePokemon = Update.renamePokemonInSaveData;
+            renamePokemon(saveData, 'Bulbasaur (clone)', 'Bulbasaur (Clone)');
```
Comment (Jaaslet) [inline src/scripts/Update.ts:958]: All this is not needed, as the name of the pokemon are no longer in the savefile

Comment (RedSparr0w) [reply src/scripts/Update.ts:958]: AFAIK, it's still used in the hatchery/queue

Comment (Jaaslet) [reply src/scripts/Update.ts:958]: Oh, you are totally right... Sorry!

File: src/scripts/Update.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (CorgiOnNeptune) [reply src/scripts/Update.ts:935]: I tried a couple other things, but couldn't get a different solution that worked, so feel free to take a shot at it. And weird, Deoxys (Attack) was removed on any saves I tested with.
Also note I'm including the Poké Ball renames in this function as well from #2691...

[the remark this answers is not in the document]
Comment (Jaaslet) [reply src/scripts/Update.ts:935]: Yeah, let's keep it. It also fixes X-items and Dowsing Machine, for people who have not played through the BF for the last couple of patches

**Review by Jaaslet — APPROVED** [review]
It looks good. Let's merged it. I will try to merge the pokeball one today, to make sure they are both released at the same time.
Thanks for the PR

File: src/scripts/Update.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (CorgiOnNeptune) [reply src/scripts/Update.ts:935]: Cool, with the full filter, it also prevents people from having to start a new run to obtain unearned milestones if they're starting from a checkpoint.

**Pull request merged**